libs/dh/market-participant/actors/data-access-api/src/lib/dh-market-party-credentials.store.ts

```diff
@@ -119,6 +135,39 @@ export class DhMarketParticipantCertificateStore extends ComponentStore<Certific
       )
   );
 
+  readonly generateClientSecret = this.effect(
+    (
+      trigger$: Observable<{
+        actorId: string;
+        onSuccess: () => void;
+        onError: () => void;
+      }>
+    ) =>
+      trigger$.pipe(
+        tap(() => this.patchState({ generateSecretInProgress: true })),
+        exhaustMap(({ actorId, onSuccess, onError }) =>
+          this.httpClient.v1MarketParticipantActorRequestClientSecretCredentialsPost(actorId).pipe(
```

Comment:
Question: How come this is not implemented as a mutation in graphQL?

Reply:
Most likely because it was easier to expose these endpoints as REST due to the upload certificate endpoint being REST too.